engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1976,92 +1965,244 @@ public void storageMigration(final String vmUuid, final StoragePool destPool) {
     private void orchestrateStorageMigration(final String vmUuid, final StoragePool destPool) {
         final VMInstanceVO vm = _vmDao.findByUuid(vmUuid);
 
+        preStorageMigrationStateCheck(destPool, vm);
+
+        try {
+            if(s_logger.isDebugEnabled()) {
```

Comment:
do you really need to check if debug log level is enabled? You can simply log it, and let the logging framework to manage it.

Reply:
yes, you do if you are passing dynamic parameters

Reply:
Does the logging framework not allow for lambda expressions to be passed in? That would be the ideal way to handle dynamic parameters without have to put an if statement in.

Reply:
@rafaelweingartner @mike-tutkowski removed the log level check and formatted the dynamic string prior to sending it as Log message 
String msg = String.format("Offline migration of %s vm %s with volumes", vm.getHypervisorType().toString(), vm.getInstanceName());
             s_logger.debug(msg);

Reply:
@mike-tutkowski the current log4j we use does not allow that. However, log4j-2 does. I am trying to make time to finish the upgrade.

@dhlaluku even though I think that using the check `s_logger.isDebugEnabled()` is foolish in our context, you do not need to remove just because of me. I understand @DaanHoogland perspective.